server/src/main/java/io/crate/planner/SubqueryPlanner.java

```diff
@@ -43,11 +44,12 @@ public SubqueryPlanner(Function<SelectSymbol, LogicalPlan> planSubSelects) {
 
     public record SubQueries(Map<LogicalPlan, SelectSymbol> uncorrelated, Map<SelectSymbol, LogicalPlan> correlated) {
 
-        public LogicalPlan applyCorrelatedJoin(LogicalPlan source) {
+        public LogicalPlan applyCorrelatedJoin(LogicalPlan source, IntSupplier ids) {
```

Comment:
I am trying to use IntSupplier `() -> Int` instead of the PlannerContext, as it's simply more lightweight and less hassle to build e.g. in tests.